### ISSUE TITLE
Complete support semi join

### ISSUE THREAD

**Issue by cmdares** [crate/crate#7335]
two statements for semi join use case:
1:
```sql
SELECT
    COUNT(0)
FROM
    TABLE_A A
WHERE
    EXISTS (
        SELECT
            1
        FROM
            TABLE_B B
        WHERE
            A.FIELD1 = B.FIELD1
            AND A.FIELD2 = B.FIELD2
            AND B.FIELD3 = 'X'
    )
    AND A.FIELD3 IN ('A', 'B', 'C')
```

2:
```sql
SELECT
    COUNT(0)
FROM
    TABLE_A A
WHERE
    A.FIELD3 IN ('A', 'B', 'C')
    AND (A.FIELD1, A.FIELD2) IN (
        SELECT
            B.FIELD1,
            B.FIELD2
        FROM
            TABLE_B
        WHERE
            B.FIELD3 = 'X'
    )
```

It‘s can not use in current version.
crate 3.0.

**Comment by cmdares** [comment]
+1

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by proddata** [comment]
@seut 

Wondering what is missing here.

```sql
CREATE TABLE a (f1 TEXT, f2 TEXT, f3 TEXT);                                                                                                                         
CREATE TABLE b (f1 TEXT, f2 TEXT, f3 TEXT);                                                                                                                         
INSERT INTO a VALUES ('a','b','c');                                                                                                                                 
INSERT INTO b VALUES ('a','b','c');                                                                                                                                 
```

The first query should work in CrateDB

```sql
cr> SELECT count(*) FROM a WHERE EXISTS(SELECT 1 FROM b where a.f1 = b.f1 and a.f2 = b.f2 and b.f3 ='c') and a.f3 IN ('a','b','c');                                     
UnsupportedFeatureException[Can't handle Symbol [SimpleReference: f3]]
```
^^^ but this seems more like a bug, as it works fine with a subquery:

```sql
cr> WITH c AS (SELECT * FROM a WHERE EXISTS(SELECT 1 FROM b where a.f1 = b.f1 and a.f2 = b.f2 and b.f3 ='c') and a.f3 IN ('a','b','c')) 
    SELECT COUNT(*) FROM c;                                                                                                                                             
+----------+
| count(*) |
+----------+
|        1 |
+----------+
WITH 1 row in set (0.018 sec)
```

---

The second query doesn't work, as CrateDB doesn't support comparing tuples 

```sql
SELECT 1 WHERE (1,2) = (1,2);
```

(works in pg sql btw)

**Comment by seut** [comment]
@proddata 
Right, by implementing correlated subqueries and the `EXISTS` expression support,  `SEMI` and `ANTI` `JOINS` are supported as well.
Thanks for the heads up!

I'll look into the bug. Btw. it only happens once a column is used inside the filter which isn't part of the select items.

For the tuple support, I've created a dedicated issue #13419.

**Comment by seut** [comment]
> Right, by implementing correlated subqueries (https://github.com/crate/crate/issues/12887) and the EXISTS expression (https://github.com/crate/crate/pull/12900) support, SEMI and ANTI JOINS are supported as well.
> Thanks for the heads up!
>
> I'll look into the bug. Btw. it only happens once a column is used inside the filter which isn't part of the select items.

This issue is fixed now by #13668 and will be released within the next 5.2 hotfix release.

**Comment by seut** [comment]
As the only remaining tuple issue is tracked by a dedicated issue #13419, I'll close this one now.
